sdks/python/apache_beam/runners/dataflow/internal/apiclient.py

```diff
@@ -660,35 +665,42 @@ def _stage_resources(self, pipeline, options):
               'Found duplicated artifact sha256: %s (%s)',
               type_payload.path,
               type_payload.sha256)
-          staged_name = staged_hashes[type_payload.sha256]
-          dep.role_payload = beam_runner_api_pb2.ArtifactStagingToRolePayload(
-              staged_name=staged_name).SerializeToString()
+          remote_name = staged_hashes[type_payload.sha256]
+          if is_staged_role:
+            # We should not be overriding this, as dep.role_payload.staged_name
+            # refers to the desired name on the worker, whereas staged_name
+            # refers to its placement in a distributed filesystem.
```

Comment:
I'm a little confused by this comment - are you saying we're currently doing the wrong thing here? If so, should this be a TODO?